rd-kt/rd-gen/src/main/kotlin/com/jetbrains/rd/generator/nova/Types.kt

```diff
@@ -43,6 +43,30 @@ data class ImmutableListOfScalars(override val itemType : IScalar) : INonNullabl
 data class ImmutableListOfBindables(override val itemType : IBindable) : INonNullableBindable, IImmutableList
 
 
+interface IAttributedType : IHasItemType {
+    val attributes: Map<Lang, List<String>>
+}
+
+data class ScalarAttributedType<out T> internal constructor(override val itemType: T, override val attributes: Map<Lang, List<String>>) : IAttributedType, IScalar where T : IScalar {
+    override val name: String get() = itemType.name
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as ScalarAttributedType<*>
+
+        if (itemType != other.itemType) return false
+
+        return true
```

Comment:
It is deliberate. At least `customSerializersTrait` fun has distinct by IType. And in the end, we have several identical serializers and compilation errors.

So if you can suggest another solution, it will be great.

Reply:
I suggest to state it in the comment, then :)